src/main/webapp/app/exercises/quiz/manage/drag-and-drop-question/drag-and-drop-question-edit.component.ts

```diff
@@ -105,8 +106,8 @@ export class DragAndDropQuestionEditComponent implements OnInit, OnChanges, Afte
      */
     mouse: DragAndDropMouseEvent;
 
-    hintCommand = new HintCommand();
-    explanationCommand = new ExplanationCommand();
+    hintCommand = new HintCommand(this.translateService);
```

Comment:
From an object design point of view, this is not a good solution. A command should not depend on a service. We should implement this differently!

Reply:
Instead, when inserting the text in the Angular component to the editor, the text should be translated.
However, I have one issue. The user interface language of Artemis does not necessarily correspond to the language in the course. A user could prefer the English user interface, but might create an exercise in German or the other way round.

I think, we should instead add a language enum (German, English) to the Course class. And depending on its value the language of the exercise content (example) is determined

Reply:
I agree that UI language and course language do not necessarily have to be the same. The translation here is limited to the creation instructions of quiz questions. I'm not sure if it would be the right way to bind the translation of this instructions to the course language. To stick with your example, if the user prefers the English user interface and is preparing a question for a German course, the instructions should appear in English, as the instruction language does not affect the quiz question presented to the students later on. At least as long as the creator does not save the default state of the quiz question but this would be an edge case that is due to the implementation of the quiz instructions and would have the same result with the current implementation in production.
In normal use, the creator replaces the instructions with the corresponding individual strings (question, hint, explanation). Here, the creator determines the language. 
In my opinion, the chosen UI language should determine the language of the quiz instructions, just like it is the case for many other elements on exercise creation page like buttons or tooltips.

Reply:
I completely disagree. The idea of the default examples is that they reflect the course language.